### PR TITLE
Implemented Task2 feature via Cuisine option

### PR DESCRIPTION
**Pull request by kaelvergara**
* Added unit test for Condition and newly added Checkbox component

### CODE REVIEW

File: server/lib/placeHelper.js [hunk 1]
```diff
@@ -7,6 +7,10 @@ export function toSearchPlacesParams(params) {
     limit: 50,
     open_now: true,
   };
+
+  if (params.categories) {
```
Comment (daisyxu0924) [inline server/lib/placeHelper.js]: why do you need to use reduce here instead of a join?

File: client/reducers/conditionReducer.js [hunk 2]
```diff
@@ -13,6 +14,13 @@ const conditionReducer = handleActions(
       const { latitude, longitude } = action.payload;
       return { ...state, latitude, longitude };
     },
+    [actionTypes.ADD_CATEGORY](state, action) {
+      return { ...state, categories: [...state.categories, action.payload] };
+    },
+    [actionTypes.REMOVE_CATEGORY](state, action) {
+      console.log(action);
+      return { ...state, categories: [...state.categories.slice(0, action.payload), ...state.categories.slice(action.payload + 1)] };
```
Comment (daisyxu0924) [inline client/reducers/conditionReducer.js]: hm not super important but maybe you can just use one slice and spread operator for better efficiency

File: client/containers/HomePage.js [hunk 3]
```diff
@@ -13,17 +13,35 @@ class HomePage extends Component {
...
     this.props.setRadius(value);
   }
+
+  handleOnSetCategories = (category) => {
+    const { categories } = this.props.condition;
+    // use indexOf() over includes() for backwards compatibility
+    const categoryIndex = categories.indexOf(category);
+    const isCategoryAlreadyAdded = categoryIndex > -1;
+    if (isCategoryAlreadyAdded) {
+      this.props.removeCategory(categoryIndex);
```
Comment (daisyxu0924) [inline client/containers/HomePage.js]: see the inconsistency between removeCategory and addCategory? one is expecting an index and the latter is expecting a value. For cases like we should really keep consistency by passing in only one type. And if we are doing that we can just have one reducer function called toggleCategory and move all the logic to the reducer. In this case it's a lot easier to test reducer too. It's not very easy to test logic in component

File: client/reducers/conditionReducer.js (hunk 2 above)
Comment (kaelvergara) [reply client/reducers/conditionReducer.js]: I'm not really sure as I just copied this from redux guide: https://redux.js.org/recipes/structuring-reducers/immutable-update-patterns#inserting-and-removing-items-in-arrays

File: client/containers/HomePage.js (hunk 3 above)
Comment (kaelvergara) [reply client/containers/HomePage.js]: I agree, the following changes has been made:

* Merge add and remove to toggle category
* Moved toggleCategory logic from component to reducer
* Added ConditionReducer test

**Pull request merged**